### ISSUE TITLE
Microsoft.Net.Compilers.props should work for MSBuild 15

### ISSUE THREAD

**Issue by rainersigwald** [dotnet/roslyn#11344]
[Microsoft.Net.Compilers.props](https://github.com/dotnet/roslyn/blob/56f605c41915317ccdb925f66974ee52282609e7/build/NuGetAdditionalFiles/Microsoft.Net.Compilers.props#L21-L22) has a hardcoded check for `MSBuildToolsVersion` 14.0:

```xml
<!-- By default don't use the compiler server on VS 2015 and never use it
     otherwise (it is not yet supported on VS 2013 or below).-->
<UseSharedCompilation Condition="'$(UseSharedCompilation)' == ''
    or '$(MSBuildToolsVersion)' != '14.0'">false</UseSharedCompilation>
```

It seems like this should be
```xml
<UseSharedCompilation Condition="'$(UseSharedCompilation)' == ''
    or '$(MSBuildToolsVersion)' <= '14.0'">false</UseSharedCompilation>
```

(version `<=` instead of `!=`)

So that the default remains correct now that MSBuild is introducing the 15.0 version for dev15.

**Comment by mmjb** [comment]
The check for the two UsingTask statements in the file should also be updated similarly.

**Comment by jasonmalinowski** [comment]
This was fixed in #11998.
